lib/segment/src/index/hnsw_index/gpu/shaders/shared_buffer.comp

```diff
@@ -78,7 +80,8 @@ void sort(uint offset, uint count) {
     }
     subgroupMemoryBarrierShared();
 
-    for (uint k = 64; (k >> 1) < count; k <<= 1) {
+    // Start bitonic sort. Start from `k = 2 * SUBGROUP_SIZE` instead of `1` because we have already sorted blocks of size `SUBGROUP_SIZE`.
+    for (uint k = 2 * SUBGROUP_SIZE; (k >> 1) < count; k <<= 1) {
```

Comment:
`k` changed from `64` to `2 * SUBGROUP_SIZE`.

With a default subgroup size of 32 that was valid. But the new approach parameterized it in case the subgroup size is different.

This means that it's effectively the same on 32-group hardware. But it got improved on hardware having a different sub group size.